src/TLabel.cpp

```diff
@@ -67,9 +67,12 @@ void TLabel::setLeave(Host* pHost, const QString& func, const TEvent& args)
 
 void TLabel::mousePressEvent(QMouseEvent* event)
 {
-    if (event->button() == Qt::LeftButton) {
+    if (event->button() == Qt::LeftButton || event->button() == Qt::RightButton) {
         if (mpHost) {
-            mpHost->getLuaInterpreter()->callEventHandler(mClick, mClickParams);
+            TEvent mClickParamsWButton = mClickParams;
+            mClickParamsWButton.mArgumentList.prepend(event->button() == Qt::LeftButton ? "LeftButton" : "RightButton");
```

Comment:
Following the comments you just need to change this and the next to use `append` rather than `prepend` however with my comment about handling ALL buttons you are going to have to switch to a `switch` to handle `Qt::LeftButton`, `Qt::RightButton`, `Qt::MiddleButton`, `Qt::BackButton`, `Qt::ForwardButton`, `Qt::TaskButton`, `Qt::ExtraButton3` ... `Qt::ExtraButton24`

Although you _might_ get away with, I guess, a `QStringLiteral("ExtraButtonHexCode%1").arg(static_cast<unsigned int>(event->button()))` for a `default:` for the higher ones and allow the very small number (if any) who are interested - presumably they would be using some _very_ unusual kit - who have them, to decode the 2^n numbers involved themselves!

Yeah, that reminds me, if using raw strings as `QString`s in the code please wrap them with `QLatin1String(`...`)` or `QStringLiteral(`...`)` as appropriate in this class (or `tr(`...`)` if they are going to become part of the _translatable_ GUI in other classes) - as it will reduce the clean up work later on as we proceed to get Mudlet ready for I18n for the 4.0 release...